crates/solidity/testing/snapshots/cst_output/ContractDefinition/recovery_testbed/generated/0.5.0-failure.yml

```
@@ -47,62 +47,78 @@ Tree:
+                                                          - VariableDeclarationType (Rule): # 113..123 "\t\t\tuint256"
+                                                              - TypeName (Rule): # 113..123 "\t\t\tuint256"
+                                                                  - ElementaryType (Rule): # 113..123 "\t\t\tuint256"
+                                                                      - UintKeyword (Token): "uint256" # 116..123
+                                                          - Identifier (Token): "a" # 124..125
+                                                          - VariableDeclarationValue (Rule): # 125..129 " = 1"
+                                                              - Equal (Token): "=" # 126..127
+                                                              - Expression (Rule): # 127..129 " 1"
+                                                                  - DecimalNumberExpression (Rule): # 127..129 " 1"
+                                                                      - DecimalLiteral (Token): "1" # 128..129
+                                                          - Semicolon (Token): ";" # 129..130
+                                              - CloseBrace (Token): "}" # 133..134
+                          - CloseBrace (Token): "}" # 136..137
+          - ContractMember (Rule): # 138..210 "\n\tfunction abc() {\n\t\tuint256 x = 0;\n\t\tunchecked { ..."
+              - FunctionDefinition (Rule): # 138..210 "\n\tfunction abc() {\n\t\tuint256 x = 0;\n\t\tunchecked { ..."
+                  - FunctionKeyword (Token): "function" # 140..148
+                  - FunctionName (Rule): # 148..152 " abc"
+                      - Identifier (Token): "abc" # 149..152
+                  - ParametersDeclaration (Rule): # 152..154 "()"
+                      - OpenParen (Token): "(" # 152..153
+                      - CloseParen (Token): ")" # 153..154
+                  - FunctionBody (Rule): # 154..210 " {\n\t\tuint256 x = 0;\n\t\tunchecked { invalid sequence..."
+                      - Block (Rule): # 154..210 " {\n\t\tuint256 x = 0;\n\t\tunchecked { invalid sequence..."
+                          - OpenBrace (Token): "{" # 155..156
+                          - Statements (Rule): # 157..174 "\t\tuint256 x = 0;\n"
+                              - Statement (Rule): # 157..174 "\t\tuint256 x = 0;\n"
+                                  - VariableDeclarationStatement (Rule): # 157..174 "\t\tuint256 x = 0;\n"
+                                      - VariableDeclarationType (Rule): # 157..166 "\t\tuint256"
+                                          - TypeName (Rule): # 157..166 "\t\tuint256"
+                                              - ElementaryType (Rule): # 157..166 "\t\tuint256"
+                                                  - UintKeyword (Token): "uint256" # 159..166
```

One idea we discussed before when we talked about this is flattening the tree structure of the tests a little bit to make it more readable/usable. So, for the below example:

- Instead of each node on a separate line, we can group rules that have only one child (that is also a rule) into the same line, as they will have the same exact range and preview comment.
- Removing `(Rule)` and `(Token)` suffixes, since we already make the dinstinction through the YAML value/LHS on the same line.

Just putting the idea out here. Definetely not blocking for this PR, as we probably should do it in a subsequent PR to make it easier to review.

```suggestion
                                      - VariableDeclarationType -> TypeName -> ElementaryType: # 157..166 "\t\tuint256"
                                          - UintKeyword: "uint256" # 159..166
```

---

I'll add a TODO for myself.